lib/vdsm/virt/livemerge.py

```diff
@@ -431,18 +431,28 @@ def _base_needs_extend(self, drive, job, base_info):
         # Measure the required base size for this merge, based on the current
         # allocation in top. This measurement is racy, since the guest may
         # write data to top after the measure.
-        measure = self._vm.measure(
+        top_measure = self._vm.measure(
             drive.domainID,
             drive.imageID,
             job.top,
             base_info["format"],
             baseID=job.base)
+        # There may be untracked bitmaps in base if they were added after the
+        # creating the top volume. Even if this is a rare case, we need to
+        # measure the base volume to make sure we account for the bitmaps size.
+        # This may allocate more than needed.
+        base_measure = self._vm.measure(
+            drive.domainID,
+            drive.imageID,
+            job.base,
+            base_info["format"])
 
         # Consider the required size and possible bitmaps. Since some of the
         # bitmaps already exist in base, this allocates more space than needed,
         # but bitmaps are small so this is good enough.
-        required_size = measure["required"] + measure.get("bitmaps", 0)
-
+        required_size = (top_measure["required"]
+                         + top_measure.get("bitmaps", 0)
+                         + base_measure.get("bitmaps", 0))
```

Comment:
This looks like the code we have in cold merge. Can we have a generic helper measuring
top and base paths, and use it from both cold and live merge?